File: movie-ticket-system/go-without-multithreading/entities/booking.go
package entities

import "app/movie-ticket-system/go-without-multithreading/enums"

// Booking structurally packages transaction states directly tied to a User and precise Seats.
type Booking struct {
	ID         string
	User       *User
	Show       *Show
	Seats      []*Seat
	TotalPrice float64
	Status     enums.BookingStatus
}

File: movie-ticket-system/java-without-multithreading/entities/Booking.java
package entities;
import java.util.List;
import enums.BookingStatus;

public class Booking {
    private String id;
    private User user;
    private Show show;
    private List<Seat> seats;
    private double totalPrice;
    private BookingStatus status;

    public Booking(String id, User user, Show show, List<Seat> seats, double totalPrice, BookingStatus status) {
        this.id = id;
        this.user = user;
        this.show = show;
        this.seats = seats;
        this.totalPrice = totalPrice;
        this.status = status;
    }

    public String getId() { return id; }
    public BookingStatus getStatus() { return status; }
    public void setStatus(BookingStatus status) { this.status = status; }
    public double getTotalPrice() { return totalPrice; }
    public List<Seat> getSeats() { return seats; }
}
